app/assets/javascripts/components/volunteers/register_modal.jsx

```diff
@@ -0,0 +1,141 @@
+  componentWillReceiveProps(nextProps) {
+    this.state = nextProps.initData;
+  }
+
+  _updateState(e) {
+    let target = $(e.target);
+    this.state[target.attr('name')] = target.val();
+    this._nextStep(this.state, "basicForm", false);
+  }
+
+  open(e) {
+    this.setState({ showModal: true });
+  }
+
+  close(e) {
+    this.setState({ showModal: false });
+  }
+
+  _nextStep(data, key, validated, frequency) {
+    if (data && key){
+      this.setState({ [key]: data });
+    }
+    if (validated) {
+      if (key === "basicForm" && frequency && frequency === "one_time") {
+        this.setState({ step: this.state.step + 2 });
+      }
+      else {
+        this.setState({ step: this.state.step + 1 });
+      }
+    }
+  }
+
+  render() {
+    pageContent = 
+      <div className="">
+        <Modal.Body>
+          <form className="modal-pickup-form">
+            <div className="row marginTop-sm">
+              <div className={this.isEdit ? `col-md-12` : `col-md-7`}>
+                <fieldset className="input-container">
+                  <label htmlFor="name" className="label label--newline">Name (First Last)</label>
+                  <br/>
+                  <input type="text" placeholder="ie. John Doe" className="input"
+                    value={this.state.name} name="name" id="name"
+                    onChange={this._updateState} />
+                </fieldset>
+              </div>
+            </div>
+
+            <div className="row marginTop-sm">
+              <div className={this.isEdit ? `col-md-12` : `col-md-7`}>
+                <fieldset className="input-container">
+                  <label htmlFor="city" className="label label--newline">City</label>
+                  <br/>
+                  <input type="text" placeholder="ie. San Francisco" className="input"
+                    value={this.state.city} name="city" id="city"
+                    onChange={this._updateState} />
+                </fieldset>
+              </div>
+            </div>
+
+            <div className="row marginTop-sm">
+              <div className={this.isEdit ? `col-md-12` : `col-md-7`}>
+                <fieldset className="input-container">
+                  <label htmlFor="state" className="label label--newline">State (abbreviation)</label>
+                  <br/>
+                  <input type="text" placeholder="ie. CA" className="input"
+                    value={this.state.state} name="state" id="state"
+                    onChange={this._updateState} />
+                </fieldset>
+              </div>
+            </div>
+
+            <div className="row marginTop-sm">
+              <div className={this.isEdit ? `col-md-12` : `col-md-7`}>
+                <fieldset className="input-container name-container">
+                  <label htmlFor="skills" className="label label--newline">Skills</label>
+                  <br/>
+                  <textarea placeholder="ie. Software Development" value={this.state.skills}
```

Comment:
Instead of having volunteers list their skills, we want them to be able to check off from a list of provided skills. This can be your next task, but just making note of it now :-)